packages/rulesets/src/spectral/functions/parameter-not-using-common-types.ts

```diff
@@ -0,0 +1,39 @@
+// Verifies that parameters already defined in common-types are not being redefined.
+
+export const parameterNotUsingCommonTypes = (parameters: any, _opts: any, ctx: any) => {
+  if (parameters === null || !Array.isArray(parameters)) {
+    return []
+  }
+  if (parameters.length === 0) {
+    return []
+  }
+
+  // TODO: maybe read this from the most recent common-types/resource-management/v#/types.json
+  const commonTypesParametersNames = new Set([
+    "subscriptionId",
+    "api-version",
+    "resourceGroupName",
+    "operationId",
+    "location",
+    "managementGroupName",
+    "scope",
+    "tenantId",
+    "ifMatch",
+    "ifNoneMatch",
+  ])
+
+  const swagger = ctx?.documentInventory?.resolved
+
+  const allParams = parameters.concat(Object.values(swagger?.parameters ?? []))
```

Comment:
The swagger.parameters is global parameters, seems you need to use parameters directly

Reply:
@jianyexi I am checking for these parameter names in both the global parameters and the parameters for the paths